### PR TITLE
Some code cleanup in RUtil.lib

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

File: libr/util/lib.c [hunk 1]
```diff
@@ -73,120 +58,104 @@ R_API void *r_lib_dl_open(const char *libname) {
...
-	const char *ext = ".dylib";
 	env = r_str_append (env, ":/lib:/usr/lib:/usr/local/lib");
 #elif __UNIX__
 	char *env = r_sys_getenv ("LD_LIBRARY_PATH");
-	const char *ext = ".so";
 	env = r_str_append (env, ":/lib:/usr/lib:/usr/local/lib");
 #else
-	char *env = strdup (".:../../../../../../../windows/system32");
-	const char *ext = ".dll";
+	char *env = strdup (".:../../../../../../../../../../../../windows/system32");
```
Comment (xarkes) [inline libr/util/lib.c]: Wtf?
Use `C;\\Windows\\System32` no?

Comment (XVilka) [reply libr/util/lib.c]: No, the path can be different. It is better to use the API to get the address: https://stackoverflow.com/questions/3544185/how-to-get-the-full-path-to-system32-directory-with-c

**Comment by radare** [comment]
Yes this is one of many ugly thjngs in the windows code that i have observed. Im just keeping it as is because “it works” but its clearly crap. So i would appreciate if anybody caring about this operating system to have a look at it

**Comment by mrexodia** [comment]
I looked at Windows, basically it's *a bit* more complicated that searching the `.` (which is actually wrong on Windows, should be the application path which is different than `.`) and system32. If anyone is interested you can find a full list at: https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order

Funny thing:

> The 16-bit system directory. There is no function that obtains the path of this directory, but it is searched.

File: libr/util/lib.c (hunk 1 above)
Comment (XVilka) [reply libr/util/lib.c]: @GustavoLCR @pelijah can you please add and test a proper way to get the full path here?

**Pull request merged**